src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -603,8 +603,26 @@ def _report_single_requirement_conflict(
 
         cands = self._finder.find_all_candidates(req.project_name)
         skipped_by_requires_python = self._finder.requires_python_skipped_reasons()
-        versions = [str(v) for v in sorted({c.version for c in cands})]
 
+        versions_set: Set[CandidateVersion] = set()
+        yanked_versions_set: Set[CandidateVersion] = set()
+        for c in cands:
+            is_yanked = c.link.is_yanked if c.link else False
+            if is_yanked:
```

Comment:
```suggestion
            if c.link.is_yanked:
```

Reply:
c.link can be None (https://github.com/pypa/pip/pull/12224#discussion_r1304839846)

Reply:
Ah, right, sorry.